src/stream/src/executor/mview/materialize.rs

```diff
@@ -672,9 +790,23 @@ impl<SD: ValueRowSerde> MaterializeCache<SD> {
                                     old_row_deserialized.into_inner().into_vec();
                                 replace_if_not_null(
                                     &mut row_deserialized_vec,
-                                    new_row_deserialized,
+                                    new_row_deserialized.clone(),
                                 );
-                                let updated_row = OwnedRow::new(row_deserialized_vec);
+                                let mut updated_row = OwnedRow::new(row_deserialized_vec);
+
+                                // Apply TOAST column fix for CDC tables with TOAST columns
+                                if let Some(toastable_indices) = toastable_column_indices {
+                                    // Note: we need to use old_row_deserialized again, but it was moved above
+                                    // So we re-deserialize the old row
+                                    let old_row_deserialized_again =
+                                        row_serde.deserializer.deserialize(old_row.row.clone())?;
+                                    updated_row = handle_toast_columns_for_cdc(
```

Comment:
since we will scan row to row for possible toast columns, I think we need a bench for related perf loss.